Jenkinsfile

```diff
@@ -20,6 +20,12 @@ node {
             testApp(image: img, runArgs: "-u root -e TEST_DATABASE_URL=${databaseUrl} -e CODECOV_TOKEN=${credentials('LMS_CODECOV_TOKEN')}") {
                 sh 'apk add build-base postgresql-dev python3-dev yarn'
                 sh 'pip3 install -q tox>=3.8.0'
+
+                // Hack to work around a race condition and make sure that
+                // .tox/.tox/bin/tox gets created before any of the parallelised
+                // `make` commands below try to access it.
+                sh 'cd /var/lib/lms && tox -e py36 --notest'
```

Comment:
Tox installs the latest version of itself at `.tox/.tox/bin/tox` and that copy of tox is what actually runs your testenvs. (This is tox's "auto-provisioning", it means you don't need to upgrade to newer versions of tox.) When `make` kicks off multiple `tox` commands at once, I think they were racing each other and one of the tox processes would often attempt to use `.tox/.tox/bin/tox` before another one of the tox processes had finished installing it, and would crash.

This command runs a tox testenv with no commands and no deps, just to induce it to create `.tox/.tox/bin/tox` before we run the concurrent `make` command below.